app/views/directives/add-secret-to-application.html

```diff
@@ -0,0 +1,90 @@
+<div>
+  <div class="dialog-title">
+    <h3>Add to Application</h3>
+  </div>
+  <div class="dialog-body">
+    <form name="addToApplicationForm" novalidate>
+      <fieldset ng-disabled="disableInputs">
+        <div class="form-group">
+          <label>Add this secret to application:</label>
+          <div class="application-select">
+            <ui-select ng-model="ctrl.application" required="true" ng-disabled="ctrl.disableInputs">
+              <ui-select-match placeholder="{{ctrl.applications.length ? 'Select an application' : 'There are no applications in this project'}}">
+              <span>
+                {{$select.selected.metadata.name}}
+                <small class="text-muted">&ndash; {{$select.selected.kind | humanizeKind : true}}</small>
+              </span>
+              </ui-select-match>
+              <ui-select-choices
+                repeat="application in (ctrl.applications) | filter : { metadata: { name: $select.search } } track by (application | uid)"
+                group-by="ctrl.groupByKind">
+                <span ng-bind-html="application.metadata.name | highlight : $select.search"></span>
+              </ui-select-choices>
+            </ui-select>
+          </div>
+        </div>
+        <div class="form-group">
+          <label>Add secret as:</label>
```

Comment:
Semantically, this would be better as a legend element inside of a fieldset.  Not sure we have styling for that...